pkg/policy/validate.go

```diff
@@ -1070,3 +1078,51 @@ func validateKinds(kinds []string, mock bool, client dclient.Interface, p kyvern
 	}
 	return nil
 }
+
+func validateNamespaces(s *kyvernov1.Spec, path *field.Path) error {
+	action := map[string]sets.String{
+		string(kyvernov1.Enforce): sets.NewString(),
+		string(kyvernov1.Audit):   sets.NewString(),
+		"enforceW":                sets.NewString(),
+		"auditW":                  sets.NewString(),
+	}
+
+	for i, vfa := range s.ValidationFailureActionOverrides {
+		patternList, nsList := utils.SeperateWildcards(vfa.Namespaces)
+
+		if vfa.Action == kyvernov1.Audit {
+			if action[string(kyvernov1.Enforce)].HasAny(nsList...) {
+				return fmt.Errorf("conflicting namespaces found in path: %s: %s", path.Index(i).Child("namespaces").String(),
+					strings.Join(action[string(kyvernov1.Enforce)].Intersection(sets.NewString(nsList...)).List(), ", "))
+			}
+			action["auditW"].Insert(patternList...)
+		} else if vfa.Action == kyvernov1.Enforce {
+			if action[string(kyvernov1.Audit)].HasAny(nsList...) {
+				return fmt.Errorf("conflicting namespaces found in path: %s: %s", path.Index(i).Child("namespaces").String(),
+					strings.Join(action[string(kyvernov1.Audit)].Intersection(sets.NewString(nsList...)).List(), ", "))
+			}
+			action["enforceW"].Insert(patternList...)
+		}
+		action[string(vfa.Action)].Insert(nsList...)
+
+		pat, ns, notOk := utils.CheckWildcardNamespaces(action["auditW"].List(), action[string(kyvernov1.Enforce)].List())
+		if notOk {
+			return fmt.Errorf("path: %s: wildcard pattern '%s' matches with namespace '%s'", path.Index(i).Child("namespaces").String(), pat, ns)
+		}
+		pat, ns, notOk = utils.CheckWildcardNamespaces(action["enforceW"].List(), action[string(kyvernov1.Audit)].List())
+		if notOk {
+			return fmt.Errorf("path: %s: wildcard pattern '%s' matches with namespace '%s'", path.Index(i).Child("namespaces").String(), pat, ns)
+		}
+
+		pat1, pat2, notOk := utils.CheckWildcardNamespaces(action["auditW"].List(), action["enforceW"].List())
+		if notOk {
+			return fmt.Errorf("path: %s: wildcard pattern '%s' conflicts with the pattern '%s'", path.Index(i).Child("namespaces").String(), pat1, pat2)
+		}
+		pat1, pat2, notOk = utils.CheckWildcardNamespaces(action["enforceW"].List(), action["auditW"].List())
+		if notOk {
+			return fmt.Errorf("path: %s: wildcard pattern '%s' conflicts with the pattern '%s'", path.Index(i).Child("namespaces").String(), pat1, pat2)
+		}
```

Comment:
Can we simplify this logic or at least make it more readable?

Reply:
@ansalamdaniel we can use small function blocks to replace these condition blocks (if-else). Like instead of four "CheckwildcardNamespaces" function calls, use one function that deals with all the logic of checking wildcards. You can return "if notOk" errors from there to the caller function and handle them all at once.

I've approved the PR but please address this change before creating upstream PR.

Reply:
@samkulkarni20 @ApsTomar I've addressed the review comments.. Please take a look..

Reply:
@ansalamdaniel Looks better now.